datajoint/connection.py

```diff
@@ -254,13 +257,12 @@ def set_query_cache(self, query_cache=None):
     def purge_query_cache(self):
         """Purges all query cache."""
         if (
-            "query_cache" in config
-            and isinstance(config["query_cache"], str)
-            and pathlib.Path(config["query_cache"]).is_dir()
+            isinstance(config.get(cache_key), str)
+            and pathlib.Path(config[cache_key]).is_dir()
         ):
-            path_iter = pathlib.Path(config["query_cache"]).glob("**/*")
-            for path in path_iter:
-                path.unlink()
+            for path in pathlib.Path(config[cache_key]).iterdir():
```

Comment:
Thanks @dimitri-yatsenko. I do recall now when this change was made. After discussion, you mentioned that the previous implementation did not delete them in the right order. Why does deleting them in a certain order matter if we are purging all the cache anyway?

Reply:
The order needs to be depth first, starting from the leaf nodes going up. You don't want to delete the folder before deleting all its files first.